src/arrayMethodJoin.js

```diff
@@ -3,9 +3,34 @@
 /**
  * Implement method join
  */
+
+const isNotIgnoredEl = (el) => ![null, undefined].includes(el);
+
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const currentArray = this;
+
+    if (!(currentArray instanceof Array)) {
+      return '';
+    }
```

Comment:
On line 10 you create this method as a method of `Array`. So this method can be called only on array